### PR TITLE
fix: persist login state across refreshes

### PR DESCRIPTION
**Pull request by moldy530**
# This PR

I had to move a number of things around to enable the cookie storage on the app so that's most of this PR. That was actually the only thing missing to remove the loading flicker. I also refactored the Mint Card slightly because that was also causing a flicker. Now it always renders the mint card if you're logged in, just disables the button until minting is ready.

---

## PR-Codex overview
This PR focuses on updating the configuration and functionality of the `AuthCardHeader`, `Providers`, and various components to enhance the authentication process and state management. It also upgrades the versioning of the configuration and related components.

### Detailed summary
- Updated `version` from `8` to `9` in multiple files.
- Enhanced `AuthCardHeader` to accept new props.
- Modified `Providers` to include `initialState` prop.
- Integrated `alchemyConfig` and `initialState` into `RootLayout`.
- Refactored `MintCard` to handle loading states better.
- Adjusted imports and exports for better modularity.
- Simplified `AuthCardWrapper` by removing unnecessary client checks.
- Added cookie handling in `RootLayout` for state initialization.
- Updated configuration structure in `config.tsx`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: examples/ui-demo/src/app/config.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  };
+  supportUrl?: string;
+};
+
+export const DEFAULT_CONFIG: Config = {
+  auth: {
+    showEmail: true,
+    showExternalWallets: false,
+    showPasskey: true,
+    addPasskey: true,
```
Comment (rthomare) [inline examples/ui-demo/src/app/config.tsx:43]: this is false right?

Comment (rthomare) [reply examples/ui-demo/src/app/config.tsx:43]: or will be...

**Review by rthomare — APPROVED** [review]

File: examples/ui-demo/src/app/config.tsx (hunk 1 above)
Comment (rthomare) [reply examples/ui-demo/src/app/config.tsx:43]: I guess fine to do it in another PR!

Comment (moldy530) [reply examples/ui-demo/src/app/config.tsx:43]: ah yea I think that's what Iyk and Rob are gonna tackle changing

**Pull request merged**